lib/features/email/presentation/email_view.dart

```diff
@@ -281,54 +283,55 @@ class EmailView extends GetWidget<SingleEmailController> with AppLoaderMixin {
-              if (controller.calendarEvent.value != null) {
-                return CalendarEventActionBannerWidget(
-                  calendarEvent: controller.calendarEvent.value!,
-                  listFromEmailAddress: controller.currentEmail?.from
-                );
-              } else {
-                return const SizedBox.shrink();
-              }
-            }),
-            if (PlatformInfo.isWeb)
-              Expanded(child: Padding(
-                padding: const EdgeInsetsDirectional.only(start: 16, bottom: 16),
-                child: _buildEmailContent(context, constraints, email)
-              ))
+            _buildLoadingContentView(email),
+            if (email.hasCalendarEvent)
+              Obx(() {
+                if (controller.calendarEvent.value != null) {
+                  return Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    mainAxisSize: MainAxisSize.min,
+                    children: [
+                      CalendarEventInformationWidget(
+                        calendarEvent: controller.calendarEvent.value!
+                      ),
+                      CalendarEventActionBannerWidget(
+                        calendarEvent: controller.calendarEvent.value!,
+                        listFromEmailAddress: controller.currentEmail?.from
+                      ),
+                      CalendarEventDetailWidget(
+                        calendarEvent: controller.calendarEvent.value!
+                      ),
+                    ],
+                  );
+                } else {
+                  return const SizedBox.shrink();
+                }
+              })
             else
-              Padding(
-                padding: const EdgeInsetsDirectional.symmetric(
-                  vertical: EmailViewStyles.mobileContentVerticalMargin,
-                  horizontal: EmailViewStyles.mobileContentHorizontalMargin
-                ),
-                child: _buildEmailContent(context, constraints, email)
-              )
+              _buildEmailContent(context, constraints, email)
           ],
         );
       });
   }
 
-  Widget _buildLoadingContentView() {
+  Widget _buildLoadingContentView(PresentationEmail selectedEmail) {
     return Obx(() {
       return controller.viewState.value.fold(
         (failure) => const SizedBox.shrink(),
         (success) {
-          if (success is GetEmailContentLoading) {
-            return loadingWidget;
+          if (selectedEmail.hasCalendarEvent) {
+            if ((success is GetEmailContentLoading || success is ParseCalendarEventLoading)) {
+              return loadingWidget;
+            } else {
+              return const SizedBox.shrink();
+            }
           } else {
-            return const SizedBox.shrink();
+            if (success is GetEmailContentLoading) {
+              return loadingWidget;
+            } else {
+              return const SizedBox.shrink();
+            }
```

Comment:
i really want to reduce if-else nested, but don't know how. wdyt ?

Reply:
In many cases, we need to separate the if-else condition clearly so that it is easy to understand to avoid the case. To make it easier to understand we just need to separate it into functions. But the if-else condition is still the same